File: Java/Java Basico/Exceptions/teste1.java
public class Exceptions {

    // throws - avisa que pode lançar excecao
    static double dividir(double a, double b) throws ArithmeticException {
        if (b == 0) throw new ArithmeticException("Divisao por zero!");
        return a / b;
    }

    public static void main(String[] args) {

        // try/catch/finally
        try {
            double resultado = dividir(10, 0);
            System.out.println(resultado);
        } catch (ArithmeticException e) {
            System.out.println("Erro: " + e.getMessage());
        } finally {
            System.out.println("Sempre executa!");
        }

        // multiplos catch
        try {
            String s = null;
            s.length(); // NullPointerException
        } catch (NullPointerException e) {
            System.out.println("Era nulo!");
        } catch (Exception e) {
            System.out.println("Erro generico: " + e.getMessage());
        }

        // excecao customizada
        try {
            throw new IllegalArgumentException("Argumento invalido!");
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: Rust/Option/teste1.rs
fn dividir(a: f64, b: f64) -> Option<f64> {
    if b == 0.0 {
        None
    } else {
        Some(a / b)
    }
}

fn main() {
    match dividir(10.0, 2.0) {
        Some(resultado) => println!("Resultado: {}", resultado),
        None => println!("Divisao por zero!")
    }

    // jeito curto
    if let Some(r) = dividir(10.0, 0.0) {
        println!("{}", r);
    } else {
        println!("Erro!");
    }
}